lib/vmpooler/api/v1.rb

```diff
@@ -36,16 +36,46 @@ def need_token!
       validate_token(backend)
     end
 
-    def fetch_single_vm(template)
-      vm = backend.spop('vmpooler__ready__' + template)
-      return [vm, template] if vm
+    def pool_backend(pool)
```

Comment:
a similar method already exists in the helpers and used 4-5 times in the code

https://github.com/puppetlabs/vmpooler/blob/2daa5244b8f71d3ceecf6695c235e3f2b0c30dbe/lib/vmpooler/providers/vsphere.rb#L451-L459